### ISSUE TITLE
Make use of artifacts produced by executing dbt test

### ISSUE THREAD

**Issue by john-jerome** [brooklyn-data/dbt_artifacts#26]
Hey there!
After analysing dbt models from this package I came to the conclusion that all artifacts that are produced by running `dbt test` (or, simply representing tests in `manifest.json` with `resource_type = 'test'`) are not being used at all.
In all stg models either `data:args:which = 'run'` for `run_results.json` or `node.value:resource_type = 'model'` for `manifest.json`.
Hence, if my assumption above is correct, running 
```
$ dbt  test
$ dbt  run-operation upload_dbt_artifacts --args '{filenames: [run_results]}'
```
will load `run_results.json` into Snowflake but it won't be picked up by any of the consequent models (except for `stg_dbt__artifacts`, but it stops there actually). 
Long story short, I suggest adding the following to the package:
1. Adding 'which run' column (name TBD) to the source table as `data:args:which` which should allow distinguishing between artifacts produced - this can also be taken from the `data` column, so it's a nice to have.
2. Building some models to analyse the results produced by `dbt test` - initially, there can be a model called `stg_dbt__test_executions` and then some other models built on top of this one.

Ideally, none of us would want any tests failing in production. However, in real-world they sometimes do and it would be nice to know which ones, since when, some SLA-related metrics, etc.

Let me know what you think of this :)

**Comment by NiallRees** [comment]
Resolved by https://github.com/brooklyn-data/dbt_artifacts/pull/66
